src/plugins/cosmos/hooks/useStakingAction/useStakingAction.tsx

```diff
@@ -37,79 +42,59 @@ export const useStakingAction = () => {
   } = useWallet()
 
   const handleStakingAction = async (data: StakingInput) => {
-    if (wallet) {
-      try {
-        const chainId = data.asset.chainId
-        const adapter = chainAdapterManager.get(chainId) as unknown as cosmos.ChainAdapter // FIXME: this is silly
-        if (!adapter) throw new Error(`unsupported chainId ${chainId}`)
+    const { chainSpecific, validator, action, value, asset } = data
 
-        let result
+    if (!wallet) return
```

Comment:
Reorganize for early return/throws to improve readability and decrease indent depth